File: network/src/main/java/com/ayvytr/network/CacheNetworkInterceptor.java
package com.ayvytr.network;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * @author ayvytr
 */
public class CacheNetworkInterceptor implements Interceptor {
    private int maxAge;

    public CacheNetworkInterceptor() {
        this(600);
    }

    public CacheNetworkInterceptor(int maxAge) {
        this.maxAge = maxAge;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request newRequest = chain.request().newBuilder()
                                  .removeHeader("Pragma")
                                  .header("Cache-Control", "public, max-age=" + maxAge)
                                  .build();
        return chain.proceed(newRequest);
    }
}

File: network/src/main/java/com/ayvytr/network/interceptor/CacheNetworkInterceptor.kt
package com.ayvytr.network.interceptor

import com.ayvytr.network.isNetworkAvailable
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

internal class CacheNetworkInterceptor(val maxAge: Int = 3600) : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        //无缓存,进行缓存
        return chain.proceed(chain.request()).newBuilder()
            .removeHeader("Pragma")
            .addHeader("Cache-Control", "max-age=$maxAge")
            .build()
    }
}

